test/WebJobs.Script.Tests/ExtensionBundle/ExtensionBundleManagerTests.cs

```diff
@@ -384,10 +385,27 @@ public async Task GetExtensionBundle_CannotReachZipEndpoint_ReturnsFalseAsync()
             Assert.Null(await manager.GetExtensionBundlePath(httpClient));
         }
 
+        [Theory]
+        [InlineData("[3.*, 4.0.0)", "3.19.0")]
+        [InlineData("[4.*, 5.0.0)", "4.2.0")]
+        [InlineData("[4.*, 5.0.0)", null)]
+        public void LimitMaxVersion(string versionRange, string version)
+        {
+            var range = VersionRange.Parse(versionRange);
+            var resolvedVersion = ExtensionBundleManager.FindBestVersionMatch(range, new List<string>()
+            { "3.7.0", "3.10.0", "3.11.0", "3.15.0", "3.14.0", "2.16.0", "3.13.0", "3.12.0", "3.9.1", "2.12.1", "2.18.0", "3.16.0", "2.19.0", "3.17.0", "4.0.2", "2.20.0", "3.18.0", "4.1.0", "4.2.0", "2.21.0", "3.19.0", "3.19.2", "4.3.0", "3.20.0" },
+            ScriptConstants.DefaultExtensionBundleId, new FunctionsHostingConfigOptions());
+
+            if (string.IsNullOrEmpty(version))
+            {
+                Assert.Equal("4.3.0", resolvedVersion);
+            }
```

Comment:
Should we add an Assert for non-empty version also? And one more inline data for v3 - `[InlineData("[3.*, 4.0.0)", null)]`?